### PR TITLE
Add user_id_or_ids and presences to guild member chunk and improve cache behaviour.

### PR DESCRIPTION
**Pull request by FasterSpeeding**

### CODE REVIEW

File: disco/gateway/client.py [hunk 1]
```diff
@@ -266,14 +267,21 @@ def run(self):
...
             # This is simply unfortunate naming on the part of Discord...
             'guild_id': guild_id_or_ids,
-            'query': query or '',
             'limit': limit,
-        })
+        }
+        payload.update(optional(
+            # This is simply Discord sticking to an unfortunate naming scheme...
+            user_ids=user_id_or_ids,
+            query=query or '' if user_id_or_ids is None else None,
```
Comment (b1naryth1ef) [inline disco/gateway/client.py]: Whats the behavior on Discord's side if both a `query` and `user_ids` is sent? Ideally this function should raise an exception if the caller passes invalid data (e.g. both query and user_id_or_ids), or we should consider splitting this up into two functions.

File: disco/state.py [hunk 2]
```diff
@@ -285,6 +290,11 @@ def on_guild_member_add(self, event):
         if event.member.guild_id not in self.guilds:
             return
 
+        if (self.guilds[event.member.guild_id].member_count is not UNSET and
```
Comment (b1naryth1ef) [inline disco/state.py]: I'm not sure I love this behavior being built into the lib, but I'd have to think on it more. For now I'd move this to a separate PR.

File: disco/gateway/client.py (hunk 1 above)
Comment (FasterSpeeding) [reply disco/gateway/client.py]: As far as behaviour goes, Discord ignores user_ids when it is sent with query.

Comment (b1naryth1ef) [reply disco/gateway/client.py]: In that case lets either split this up into a new function, or raise an exception if both query and user_ids is passed.
